gdnative-derive/src/lib.rs

```diff
@@ -164,6 +201,49 @@ pub fn profiled(meta: TokenStream, input: TokenStream) -> TokenStream {
 /// - `no_editor`
 ///
 /// Hides the property from the editor. Does not prevent it from being sent over network or saved in storage.
+///
+/// ### `#[methods]`
+/// Adds the necessary information to a an `impl` block to register the properties to be and methods with Godot.
+///
+/// **Important**: This needs to be added to one and only one `impl` block for a given `NativeClass`
+///
+/// For additional details about how `#[methods]` expands, please refer to [gdnative::methods](methods)
+///
+/// ### `#[export]`
+/// Registers the attributed function signature to be used by Godot.
+/// A valid function signature must have
+/// - `&self` or `&mut self` as its first parameter
+/// - `&T` or `TRef<T>` where T refers to the type declared in `#[inherit(T)]` attribute as it's second parameter. This is typically called the `owner`
+/// - Additional parameter must implement the `FromVariant` trait or be a Variant type.
```

Comment:
Maybe to follow the "A valid function signature must have", we can explain that 0..N extra parameters are OK:

```
/// A valid function signature must have:
/// - `&self` or `&mut self` as its first parameter
/// - `&T` or `TRef<T>` where ...
/// - optionally any number of additional parameters, which must implement have type `Variant` 
///   or implement the `FromVariant` trait. This trait is implemented for most common types.